includes/admin/import/class-batch-import.php

```diff
@@ -0,0 +1,164 @@
+<?php
+/**
+ * Batch Import Class
+ *
+ * This is the base class for all batch import methods. Each data import type (customers, payments, etc) extend this class
+ *
+ * @package     EDD
+ * @subpackage  Admin/Import
+ * @copyright   Copyright (c) 2015, Pippin Williamson
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       2.6
+ */
+
+// Exit if accessed directly
+if ( ! defined( 'ABSPATH' ) ) exit;
+
+/**
+ * EDD_Batch_Import Class
+ *
+ * @since 2.6
+ */
+class EDD_Batch_Import {
+
+	/**
+	 * The file being imported
+	 *
+	 * @since 2.6
+	 */
+	public $file;
+
+	/**
+	 * The parsed CSV file being imported
+	 *
+	 * @since 2.6
+	 */
+	public $csv;
+
+	/**
+	 * Total rows in the CSV file
+	 *
+	 * @since 2.6
+	 */
+	public $total;
+
+	/**
+	 * The current step being processed
+	 *
+	 * @since 2.6
+	 */
+	public $step;
+
+	/**
+	 * The number of items to process per step
+	 *
+	 * @since 2.6
+	 */
+	public $per_step = 20;
+
+	/**
+	 * The capability required to import data
+	 *
+	 * @since 2.6
+	 */
+	public $capability_type = 'manage_shop_settings';
+
+	/**
+	 * Is the import file empty
+	 *
+	 * @since 2.6
+	 */
+	public $is_empty = false;
+
+	public $field_mapping = array();
+
+	/**
+	 * Get things started
+	 *
+	 * @param $_step int The step to process
+	 * @since 2.6
+	 */
+	public function __construct( $_file = '', $_step = 1 ) {
+
+		if( ! class_exists( 'parseCSV' ) ) {
```

Comment:
This should be done in the class in case extension authors directly load the class as well